pkg/network/server.go

```diff
@@ -282,6 +279,48 @@ func newServerFromConstructors(config ServerConfig, chain Ledger, stSync StateSy
 	return s, nil
 }
 
+type blockAdapter struct {
+	bc StateSync
+}
+
+func (a blockAdapter) AddItem(b *block.Block) error {
+	return a.bc.AddItem(b)
+}
+
+func (a blockAdapter) AddItems(bs ...*block.Block) error {
+	for _, b := range bs {
+		if err := a.bc.AddItem(b); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a blockAdapter) Height() uint32 {
+	return a.bc.Height()
+}
+
+type chainAdapter struct {
+	chain Ledger
+}
+
+func (a chainAdapter) AddItem(b *block.Block) error {
+	return a.chain.AddItem(b)
+}
+
+func (a chainAdapter) AddItems(bs ...*block.Block) error {
+	for _, b := range bs {
+		if err := a.chain.AddItem(b); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a chainAdapter) Height() uint32 {
+	return a.chain.Height()
```

Comment:
Ditto for height. It must be BlockHeight() for blocks adapter and HeaderHeight() for headers adapter.